File: plugins/clock/clock.1s.js
#!/usr/bin/env node
/**
 * Clock Plugin - Shows current time using Crossbar API
 */
const { execSync } = require('child_process');

function crossbar(...args) {
    try {
        return execSync(`crossbar ${args.join(' ')}`, { encoding: 'utf8', timeout: 5000 }).trim();
    } catch {
        return null;
    }
}

const now = new Date();
const timeStr = crossbar('time') || now.toLocaleTimeString('en-GB');
const dateStr = crossbar('date') || now.toISOString().split('T')[0];

console.log(`🕐 ${timeStr}`);
console.log('---');
console.log(`Time: ${timeStr}`);
console.log(`Date: ${dateStr}`);
console.log('---');
console.log('Refresh | refresh=true');

File: plugins/clock/clock.1s.dart
#!/usr/bin/env dart
/// Clock Plugin - Shows current time using Crossbar API
import 'dart:io';

String? crossbar(List<String> args) {
  try {
    final result = Process.runSync('crossbar', args);
    return result.exitCode == 0 ? (result.stdout as String).trim() : null;
  } catch (_) {
    return null;
  }
}

void main() {
  final now = DateTime.now();
  final timeStr = crossbar(['time']) ?? 
      '${now.hour.toString().padLeft(2, '0')}:${now.minute.toString().padLeft(2, '0')}:${now.second.toString().padLeft(2, '0')}';
  final dateStr = crossbar(['date']) ?? 
      '${now.year}-${now.month.toString().padLeft(2, '0')}-${now.day.toString().padLeft(2, '0')}';

  print('🕐 $timeStr');
  print('---');
  print('Time: $timeStr');
  print('Date: $dateStr');
  print('---');
  print('Refresh | refresh=true');
}
